unstructured_ingest/v2/processes/connectors/lancedb.py

```diff
@@ -0,0 +1,161 @@
+import asyncio
+import json
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Optional
+
+import pandas as pd
+from pydantic import Field, Secret
+
+from unstructured_ingest.error import DestinationConnectionError
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces.connector import AccessConfig, ConnectionConfig
+from unstructured_ingest.v2.interfaces.file_data import FileData
+from unstructured_ingest.v2.interfaces.upload_stager import UploadStager, UploadStagerConfig
+from unstructured_ingest.v2.interfaces.uploader import Uploader, UploaderConfig
+from unstructured_ingest.v2.processes.connector_registry import DestinationRegistryEntry
+
+CONNECTOR_TYPE = "lancedb"
+
+if TYPE_CHECKING:
+    from lancedb import AsyncConnection
+
+
+SUPPORTED_ELEMENT_FIELDS = (
+    "text",
+    "element_id",
+    "type",
+)
+SUPPORTED_ELEMENT_METADATA_FIELDS = (
```

Comment:
How did we choose these fields to limit to?

Reply:
Does lance have limits? 

The only reason we did this with pinecone was because it has strict limits